backends/gstreamer/src/player.rs

```diff
@@ -873,4 +881,10 @@ impl Player for GStreamerPlayer {
 
         ret
     }
+
+    fn shutdown(&self) -> Result<(), PlayerError> {
+        self.observers.lock().unwrap().clear();
+        self.renderers.lock().unwrap().clear();
+        self.stop()
```

Comment:
shouldn't stop be before the observers and renders clearing?

Reply:
Stopping will trigger shutdown related messages (i.e. state change message) that observers may not be able to handle anymore. Since this is an active request to shutdown it is safe to assume that the client already knows about the shutdown related notifications and makes the API easier to use (i.e. the client does not need to keep a shutdown flag to discard shutdown messages).

Reply:
ok!